connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -88,7 +88,8 @@ public void stop() {
     // divide consumer groups among tasks
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
-        if (!config.enabled() || knownConsumerGroups.isEmpty()) {
+        if (!config.enabled() || knownConsumerGroups.isEmpty()
+                || config.emitCheckpointsInterval().isNegative()) {
```

Comment:
It feels like allowing intervals to be negative interval to disable connectors is not a great pattern. Instead preventing the interval to be negative and relying solely on `enabled()` would have been a better alternative.

Reply:
@ryannedolan , as @mimaison suggested, if `emit.checkpoints.enabled = false`, do you think if it is a better alternative to return `Long.MAX_VALUE`, rather than negative?

```
    Duration emitCheckpointsInterval() {
        if (getBoolean(EMIT_CHECKPOINTS_ENABLED)) {
            return Duration.ofSeconds(getLong(EMIT_CHECKPOINTS_INTERVAL_SECONDS));
        } else {
            // negative interval to disable
            return Duration.ofMillis(Long.MAX_VALUE);
        }
    }
```

Reply:
All the interval configs (sync topics acls, sync topic configs, refresh topics, ...) use negative values to indicate that the Scheduler should not run the corresponding task. I suppose we could have used an Optional or something instead. If we drop the negatives here, we should do so with all the other tasks. Mox nix to me.

Reply:
It's not ideal but yes we don't want to change the current behaviour here. Let's keep the negative values.

Reply:
great, thanks for all of your inputs. Then I suppose the only change in the next update is to `adding the explanation to the comment on why we use `isNegative()` to disable the `heartbeat` and `checkpoint` emissions.